bertopic/_bertopic.py

```diff
@@ -1058,12 +1075,54 @@ def _reduce_topics(self, documents: pd.DataFrame) -> pd.DataFrame:
         Returns:
             documents: Updated dataframe with documents and the reduced number of Topics
         """
-        if isinstance(self.nr_topics, int):
-            documents = self._reduce_to_n_topics(documents)
-        elif isinstance(self.nr_topics, str):
+        self.nr_topics = max(self.nr_topics, 2)
+        if self.clustering_method == 'gmm':
+            documents = self._reduce_with_gmm(documents)
+        elif self.nr_topics == 'auto':
             documents = self._auto_reduce_topics(documents)
         else:
-            raise ValueError("nr_topics needs to be an int or 'auto'! ")
+            raise ValueError("nr_topics needs to be an int or 'auto' or 'clustering'! ")
+
+        return documents
+
+    def _reduce_with_gmm(self, documents):
+        if not self.mapped_topics:
+            self.mapped_topics = {}
+
+        initial_nr_topics = len(self.get_topics())
```

Comment:
What's the complexity of this call? If not O(1), can't we get it from a static variable?